src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -154,6 +154,15 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
             }
         }
 
+        public void PutMyNodeInfoAfterClusterVersionSetting(object sender, AfterClusterVersionSettingEventArgs args)
+        {
+            if (args.PreviousClusterVersion >= args.CurrentClusterVersion)
```

Comment:
you want to do it always or only when previous is 0?

Reply:
Considering the context provided earlier, I believe we should do it only when the previous cluster version is 0.